nixos/doc/manual/configuration/fhs.chapter.md

```diff
@@ -0,0 +1,199 @@
+hierarchy into any global paths, meaning that the user wishing to run a program
+expecting the "FHS" environment needs to explicitly point the program at its
+dependencies. NixOS provides several tools that automate or otherwise
+accommodate this task.
+
+## Dynamic linkage issues {#sec-fhs-dynamic-linkage}
+
+Dynamically linked native programs built for "FHS" environments expect to find
+the so-called "dynamic linker" in a fixed location, e.g.
+`/lib64/ld-linux-x86-64.so.2` depending on the platform. They might also expect
+a number of "system" libraries to be provided by the environment. In "FHS"
+distributions these are deployed in `/usr/lib`, and paths to them are
+discovered by the dynamic linker through the `/etc/ld.so.conf` file. These
+files are not available in NixOS by default.
+
+NixOS provides the `programs.nix-ld.enable` option which deploys a proxy where
+"FHS" distributions deploy the dynamic linker (e.g.
+`/lib64/ld-linux-x86-64.so.2`). This proxy allows the user to redirect "FHS"
+programs to the "system dependencies" they expect using the `NIX_LD` and
+`NIX_LD_LIBRARY_PATH` variables. These variables are safe in that they do not
+affect software from Nixpkgs. The `nix-ld` NixOS module also sets up default
+values for these variables.
+
+For example, with the following configuration:
+
+```nix
+  programs.nix-ld.enable = true;
+```
+
+...packages from conda can be used as follows:
+
+```bash
+$ nix-shell -p micromamba
+$ micromamba env create -n some-env python pytorch
+$ eval "$(micromamba shell hook --shell=bash)"
+$ micromamba activate some-env
+$ python
+>>> import torch
+```
+
+Without `nix-ld` it wouldn't be possible to run the conda-installed python
+interpreter, because it wouldn't know where to find the dynamic linker:
+
+```bash
+$ micromamba activate some-env
+$ python
+bash: /home/nixos/micromamba/envs/some-env/bin/python: cannot execute: required file not found
+$ patchelf --print-interpreter $(which python)
+/lib64/ld-linux-x86-64.so.2
+```
+
+Similarly, native libraries installed from binaries may be used with `nix-ld`:
+
+```bash
+$ python -m venv ~/some-venv
+$ source ~/some-venv/bin/activate
+$ pip install zmq
+$ python
+>>> import zmq
+```
+
+Without `nix-ld` the above would have failed:
+
+```bash
+$ source ~/some-venv/bin/activate
+$ python
+>>> import zmq
+...
+ImportError: libstdc++.so.6: cannot open shared object file: No such file or directory
+```
+
+A more complex scenario may be handled using ad hoc Nix shells:
+
+```nix
+# shell.nix
+
+with import <nixpkgs> { };
+
+mkShell {
+    packages = [ micromamba ];
+    NIX_LD_LIBRARY_PATH = lib.makeSearchPath "lib" [
+      addOpenGLRunpath.driverLink # "/run/opengl-driver"
+      glib.out
+      libglvnd
+      stdenv.cc.cc.lib
+    ];
+}
+```
+
+```bash
+$ nix-shell shell.nix
+$ micromamba env create -n omnimotion python=3.8
+$ eval "$(micromamba shell hook --shell=bash)"
+$ micromamba activate omnimotion
+$ micromamba install ...
+$ pip install ...
```

Comment:
This uses project-specific names and versions, which may not be appropriate? The original idea was to present an environment sufficient to run commands from the README of this repo: https://github.com/qianqianwang68/omnimotion. This would demonstrate how more complex that may depend on OpenGL and CUDA can be still handled with `nix-ld`. I am not sure if these explanations would make sense in the manual, but it's not good that I mention e.g. `libglvnd` and do not explain why?

Reply:
I think the people that want to use these types of applications and landed here, know why they want to use them. No need to do marketing here.